### PR TITLE
Offer and validate raid names

### LINKED ISSUE #3722: storage: suggest name for a new raid
If you don't fill out the name, it will default to the name "127" and the next raid you create will default to the name "126", so we might as well fill those out in the name field to indicate that.

### LINKED ISSUE #9976: storage: RAID creation throws an error when I put a name with space in  the name
Whenever I put a space in the name of a RAID array I get an error.
The RAID still gets created though.

screenshot from 2018-09-04 11-36-34

### PR DESCRIPTION
**Pull request by marusak**
Fixes #3722
Fixes #9976

### CODE REVIEW

File: test/verify/check-storage-mdraid [hunk 1]
```diff
@@ -172,12 +182,12 @@ class TestStorageMdRaid(StorageCase):
         # Stopping and starting the array will create the expected
         # symlink in /dev/md/.
         #
-        b.wait_text(info_field("Device", m.image), "/dev/md/ARR")
+        b.wait_text(info_field("Device", m.image), "/dev/md/raid0")
 
         # Partitions also get symlinks in /dev/md/, so they should
-        # have names like "/dev/md/ARR1".
+        # have names like "/dev/md/raid01".
```
Comment (mvollmer) [inline test/verify/check-storage-mdraid:188]: That would be `/dev/md/raid0p1`, I believe.

Comment (marusak) [reply test/verify/check-storage-mdraid:188]: It indeed is!
but why it was `ARR` before and now it has that one `p`?

**Comment by marusak** [comment]
Just one well known flake on [f-33/firefox](https://logs.cockpit-project.org/logs/pull-15628-20210331-124835-02817952-fedora-33-firefox/log.html#243-2)

File: test/verify/check-storage-mdraid (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-storage-mdraid]: Because `/dev/md/raid0` ends in a digit.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**